### PR TITLE
Second task form testing:  test arrays cultivator

### PR DESCRIPTION
**Pull request by qnislavnachev**

### CODE REVIEW

File: Test/test/ArrayCiltivatorTest.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import task2.ArrayCiltivator;
+import static org.junit.Assert.*;
+
+public class ArrayCiltivatorTest {
+    ArrayCiltivator array = new ArrayCiltivator();
+    int[] arr = new int[]{5,3,6};
+    int[] reversedArray = new int[]{6,3,5};
+
+    @Test
+    public void getMinElementTest() throws Exception {
```
Comment (mgenov) [inline Test/test/ArrayCiltivatorTest.java]: You are not testing printArray method where you should to see how Global State affects testing.

File: Test/test/ArrayCiltivatorTest.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        assertTrue(array.getMinElement(arr) == 3);
+    }
+
+    @Test
+    public void getSumTest() throws Exception {
```
Comment (mgenov) [inline Test/test/ArrayCiltivatorTest.java]: This is a duplication with your production code. To verify it you can use simple math: 

```java
@Test
public void getSum() {
  // 5 + 3 + 6 = 14 
  assertThat(new ArrayCultivator(new int[] {5,3,6}).getSum(), is(equalTo(14)); 
}
```

Keep in mind that extracting test data as fields of the class should happen only if required. By moving them as fields of the class your tests become hard for understanding.

File: src/task2/ArrayCiltivator.java [hunk 3]
```diff
@@ -22,9 +22,9 @@ public int getSum(int [] arr) {
         return sum;
     }
 
-    public void printArrayElements(int [] arr){
+    public int[] printArrayElements(int [] arr){
```
Comment (mgenov) [inline src/task2/ArrayCiltivator.java]: You need to test that System.out.println is invoked for each of the elements of the array.

Comment (mgenov) [inline src/task2/ArrayCiltivator.java]: https://www.google.bg/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=testing+Syste.out+java
http://stackoverflow.com/questions/1119385/junit-test-for-system-out-println
http://programmers.stackexchange.com/questions/148108/why-is-global-state-so-evil

File: Test/test/ArrayCiltivatorTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import static org.junit.Assert.*;
+
+public class ArrayCiltivatorTest {
+    ArrayCiltivator array = new ArrayCiltivator();
+    int[] arr = new int[]{5,3,6};
+    int[] reversedArray = new int[]{6,3,5};
```
Comment (EmilGeorgiev) [inline Test/test/ArrayCiltivatorTest.java]: it is better this fields to be in tests

File: src/task2/ArrayCiltivator.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
+package task2;
+
+import java.util.Arrays;
+
+public class ArrayCiltivator {
+
+    public int getMinElement(int[] arr) {
+        int min = arr[0];
```
Comment (EmilGeorgiev) [inline src/task2/ArrayCiltivator.java]: what will happen if we try to get min element from empty arrayt

File: src/task2/ArrayCiltivator.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        return sum;
+    }
+
+    public void printArrayElements(int[] arr) {
+        System.out.print(Arrays.toString(arr));
+
+
+    }
+
+    public int[] reverseArray(int[] arr) {
```
Comment (EmilGeorgiev) [inline src/task2/ArrayCiltivator.java]: There is a rule named "rule of three", which states: 
```
Rule of three is a code refactoring rule of thumb to decide when a replicated piece of code 
should be replaced by a new procedure. It states that the code can be copied once, but that 
when the same code is used three times, it should be extracted into a new procedure. The rule 
was introduced by Martin Fowler in Refactoring[1] and attributed to Don Roberts.
```
https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)


So in your case the duplicated pattern is looking like:
```
public class A {
   public void a(int[] array); 
   public void b(int[] array);
   public void c(int[] array); 
}
```
As you can see, all methods are accepting array which is duplicated. You can remove this duplication by moving the method dependency to a constructor dependency and your code will like: 

```java
public class Array {
  private final int[] array;
  public Array(int[] array) {
      this.array = array;
  }

  public int getMinElement() {
    ...
  }
  public int getSum() {
    ...
  }
}
```

File: Test/test/ArrayCiltivatorTest.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
+package test;
```
Comment (EmilGeorgiev) [inline Test/test/ArrayCiltivatorTest.java]: package should be with lower case  ```Test/test/ArrayCiltivatorTest.java``` to ```test/test/ArrayCiltivatorTest.java```

File: test/test/ArrayCiltivatorTest.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+        System.setOut(new PrintStream(out));
+    }
+
+    @After
+    public void cleanUpStreams() {
+        System.setOut(null);
+    }
+
+    @Test
+    public void getMinElementTest() throws Exception {
```
Comment (EmilGeorgiev) [inline test/test/ArrayCiltivatorTest.java]: remove ```Test``` from name

**Comment by EmilGeorgiev** [comment]
LGTM

**Pull request closed without merging**